File: frontend/src/app/pages/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient} from '@angular/common/http';
import { Urls } from './url';
import {Observable} from "rxjs";
@Injectable({
  providedIn: 'root'
})
export class UserService {

  API_URL = new Urls();
  constructor(private http: HttpClient) { }

/*  updUser (customer) {
    return this.http.post(`${this.API_URL.url}/store` , customer);
  }*/

  gtUsers ():Observable<any> {
    return this.http.get(`${this.API_URL.url}/UserService/viewUsers`);
  }
  gtUsersGroupType ():Observable<any> {
    return this.http.get(`${this.API_URL.url}/UserService/viewUsersGroupType`);
  }
  getUsersToApprove ():Observable<any> {
    return this.http.get(`${this.API_URL.url}/UserService/viewUsersToApprove`);
  }
  getUsersToApproveDelete ():Observable<any> {
    return this.http.get(`${this.API_URL.url}/UserService/viewUsersToApproveDelete`);
  }
  addUser(user):Observable<any>  {
    return  this.http.post(`${this.API_URL.url}/UserService/addUsers`, user);
  }
  updateUser(user):Observable<any> {
    return this.http.put(`${this.API_URL.url}/UserService/updateUsers` ,user);
  }
  approveUser(user):Observable<any> {
    return this.http.put(`${this.API_URL.url}/UserService/approveUser` ,user);
  }
  deleteUser(user):Observable<any> {
    return this.http.put(`${this.API_URL.url}/UserService/deleteUser` ,user);
  }
  approveDeleteUser(user):Observable<any> {
    return this.http.put(`${this.API_URL.url}/UserService/approveDeleteUser` ,user);
  }



}

File: backend/src/main/java/com/revenue/revenueCollection/WebService/UserService.java
package com.revenue.revenueCollection.WebService;


import com.revenue.revenueCollection.Interfaces.UserBal;

import com.revenue.revenueCollection.Models.ResponseMessage;
import com.revenue.revenueCollection.Models.UserGroupType;
import com.revenue.revenueCollection.Models.Users;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.List;
@RestController
@RequestMapping(path="/api/UserService/")

public class UserService {
    @Autowired
    private UserBal userBal;


    @GetMapping(path="/viewUsers",produces = "application/json")
    public List<Users> getAllUsers() {
       return userBal.viewUserDetails();
        //return Response.status(200).entity(apps).build();

    }
    @GetMapping(path="/viewUsersGroupType",produces = "application/json")
    public List<UserGroupType> getAllUsersGroupType() {
        return userBal.getUserGroupType();
        //return Response.status(200).entity(apps).build();

    }
    @GetMapping(path="/viewUsersToApprove",produces = "application/json")
    public List<Users> getUsersToapprove() {
        return userBal.getUsersToApprove();
        //return Response.status(200).entity(apps).build();

    }
    @GetMapping(path="/viewUsersToApproveDelete",produces = "application/json")
    public List<Users> getUsersToApproveDelete() {
        return userBal.getUsersToApproveDelete();
        //return Response.status(200).entity(apps).build();

    }



    @PostMapping(path="/addUsers",consumes = "application/json",produces = "application/json")
    public Response addusers(@Valid @RequestBody Users user){

        ResponseMessage response = userBal.registerUser(user);
        return Response.status(200).entity(response).build();
    }

    @PutMapping(path="/updateUsers" ,consumes = "application/json",produces = "application/json")
    public Response updateusersDetails(@Valid @RequestBody Users users) {
        System.out.println(users.userId +"$$$$$$$$$$$$$$$$$$$$$$$$$$$$");
        ResponseMessage response = userBal.updateUser(users);
        System.out.println(users.userId +"$$$$$$$$$$$$");
        return Response.status(200).entity(response).build();
    }
    @PutMapping(path="/approveUser" ,consumes = "application/json",produces = "application/json")
    public Response approveUser(@Valid @RequestBody Users users) {
        System.out.println(users.userId +"$$$$$$$$$$$$$$$$$$$$$$$$$$$$");
        ResponseMessage response = userBal.approveUsers(users);
        System.out.println(users.userId +"$$$$$$$$$$$$");
        return Response.status(200).entity(response).build();
    }
    @PutMapping(path="/deleteUser" ,consumes = "application/json",produces = "application/json")
    public Response deleteUser(@Valid @RequestBody Users users) {
        System.out.println(users.userId +"$$$$$$$$$$$$$$$$$$$$$$$$$$$$");
        ResponseMessage response = userBal.deleteUsers(users);
        System.out.println(users.userId +"$$$$$$$$$$$$");
        return Response.status(200).entity(response).build();
    }
    @PutMapping(path="/approveDeleteUser" ,consumes = "application/json",produces = "application/json")
    public Response approveDeleteUser(@Valid @RequestBody Users users) {
        System.out.println(users.userId +"$$$$$$$$$$$$$$$$$$$$$$$$$$$$");
        ResponseMessage response = userBal.approveDeleteUsers(users);
        System.out.println(users.userId +"$$$$$$$$$$$$");
        return Response.status(200).entity(response).build();
    }
}
